pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -262,6 +262,46 @@ func (a *ApplicationBackupController) handle(ctx context.Context, backup *stork_
 			if !canDelete {
 				return nil
 			}
+			// run the post exec rules if the backup is in volume stage and execRulesCompleted is not completed
```

Comment:
Should move this step before deleteBackup api? Since we decided to delete the backup, should wait till backup cancel and deletion to run the post exec rule?

Reply:
Agree, post exec rule execution doesn't required to be on hold till the backup deletion will update accordingly.